pkg/skaffold/schema/profiles.go

```diff
@@ -68,18 +68,19 @@ func checkKubeContextConsistency(isContextImmutable bool, cliContext, effectiveC
 	if err != nil {
 		return errors.Wrap(err, "getting current cluster context")
 	}
+	currentContext := kubeConfig.CurrentContext
 
 	// nothing to do
-	if effectiveContext == "" || effectiveContext == kubeConfig.CurrentContext || !isContextImmutable {
+	if effectiveContext == "" || effectiveContext == currentContext || 0 == len(contextSpecificProfiles) {
 		return nil
 	}
 
-	return fmt.Errorf("some activated profile contains kubecontext specific settings for a different than the effective kubecontext, please revise your profile activations")
+	return fmt.Errorf("profiles %q were activated by kube-context %q, but the effective kube-context is %q -- please revise your profile activations", contextSpecificProfiles, currentContext, effectiveContext)
```

Comment:
How can we reach this state when multiple profiles `contextSpecificProfiles` activated?
This displays an error message but does not suggest an actionable item for users to fix the problem

As a user, if i am new to skaffold or kubecontexts, i would be lost. Can you suggest a recommendation ?

e.g.:
```
profiles ["minikube-profile"] were activated by kube-context "minikube", but the effective kube-context is "staging" -- please revise your profile activations by either using removing `kube-context` from your config or from global defaults`
```

Reply:
This is tricky, because there are two possible causes for the error:

1. The user specified an invalid `deploy.kubeContext` in some activated profile.
2. Some profile got inadvertently activated through a profile activation via matching kube-context.

As Skaffold cannot know what was the underlying logic error, we cannot give an accurate recommendation. However, I included the yaml paths to the questionable fields. Does that sound better to you?